src/sentences/bww.rs

```diff
@@ -0,0 +1,138 @@
+use arrayvec::ArrayString;
+use nom::{
+    bytes::complete::is_not, character::complete::char, combinator::opt, number::complete::float,
+};
+
+use crate::{parse::NmeaSentence, Error, SentenceType};
+
+const MAX_LEN: usize = 64;
+
+/// BWW - Bearing - Waypoint to Waypoint
+///
+/// Bearing calculated at the FROM waypoint.
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_bww_bearing_waypoint_to_waypoint>
+///
+/// ```text
+///        1   2 3   4 5    6    7
+///        |   | |   | |    |    |
+/// $--BWW,x.x,T,x.x,M,c--c,c--c*hh<CR><LF>
+/// ```
+/// Field Number:
+/// 1. Bearing, degrees True
+/// 2. T = True
+/// 3. Bearing Degrees, Magnetic
+/// 4. M = Magnetic
+/// 5. TO Waypoint ID
+/// 6. FROM Waypoint ID
+/// 7. Checksum
+
+#[derive(Debug, PartialEq)]
+pub struct BwwData {
+    pub true_bearing: Option<f32>,
+    pub magnetic_bearing: Option<f32>,
+    pub to_waypoint_id: Option<ArrayString<MAX_LEN>>,
+    pub from_waypoint_id: Option<ArrayString<MAX_LEN>>,
+}
+
+fn do_parse_bww(i: &str) -> Result<BwwData, Error> {
+    // 1. Bearing, degrees True
+    let (i, true_bearing) = opt(float)(i)?;
+    let (i, _) = char(',')(i)?;
+    // 2. T = True
+    let (i, _) = opt(char('T'))(i)?;
+    let (i, _) = char(',')(i)?;
+
+    // 3. Bearing, degrees Magnetic
+    let (i, magnetic_bearing) = opt(float)(i)?;
+    let (i, _) = char(',')(i)?;
+    // 4. M = Magnetic
+    let (i, _) = opt(char('M'))(i)?;
+    let (i, _) = char(',')(i)?;
+
+    // 5. TO Waypoint ID
+    let (i, to_waypoint_id) = opt(is_not(","))(i)?;
+
+    let to_waypoint_id = if let Some(to_waypoint_id) = to_waypoint_id {
+        Some(
+            ArrayString::from(to_waypoint_id)
+                .map_err(|_e| Error::SentenceLength(to_waypoint_id.len()))?,
```

Comment:
This error refers to the whole Sentence Length. I believe it should be a different error as it's only referring to the waypoint string length.

- [ ] We should add e.g. `Error::ParameterLength` for such occasions.